homeassistant/components/bmw_connected_drive/binary_sensor.py

```diff
@@ -142,13 +126,18 @@ def update(self):
                 result["check_control_messages"] = cbs_list
             else:
                 result["check_control_messages"] = "OK"
-        elif self._attribute == "charging_status":
+        # device class power: On means power detected, Off means no power
+        if self._attribute == "charging_status":
+            self._attr_state = vehicle_state.charging_status in [ChargingState.CHARGING]
             result["charging_status"] = vehicle_state.charging_status.value
             result["last_charging_end_result"] = vehicle_state.last_charging_end_result
-        elif self._attribute == "connection_status":
+        # device class plug: On means device is plugged in,
+        #                    Off means device is unplugged
+        if self._attribute == "connection_status":
+            self._attr_state = vehicle_state.connection_status == "CONNECTED"
```

Comment:
Use `self._attr_is_on` not `self._attr_state`

Reply:
How is this even working now with `self._attr_state`? :thinking:

Reply:
That line was part of #53054 -> https://github.com/home-assistant/core/pull/53054/files#diff-726ee92b7a8490c22a1a15f1e5b193a32118ff67b629b319a2d919bce7a51decR116
I combined the check of the state with the data of the attributes, that's why it shows here as a difference.

As I don't have an electric car I didn't notice this in the HA frontend 😄

Reply:
It should not have been a part of #53054, please change it everywhere it's used in this file.

Reply:
You mean change all `self._attr_state` to `self._attr_is_on` in the binary_sensor file?

I can assure you that HA is showing the correct status for the other binary sensors which are available for my car with the current code.

Reply:
Yes, that is what I ment.
How is it doing that? binary sensors does not have state, they have is_on instead

Reply:
That makes sense. Will change it.